qemu/deps/win_driver_install/win_driver_install.py

```diff
@@ -0,0 +1,145 @@
+        shutil.copy(dpinstdir, "C:\dpinst.exe")
+    else:
+        print "dpinst.exe is already existed"
+
+
+# Install certificate
+def certutil(vol_utils):
+    certutil_cmd = "certutil -addstore -f TrustedPublisher %s\redhat.cer" % vol_utils
+    if not os.path.exists("C:\Windows\system32\certutil.exe"):
+        shutil.copy("%s\certutil.exe" % vol_utils, "C:\Windows\system32")
+    if not os.path.exists("C:\Windows\system32\certadm.dll"):
+        shutil.copy("%s\certadm.dll" % vol_utils, "C:\Windows\system32")
+    try:
+        os.system(certutil_cmd)
+    except Exception, err:
+        print "Can not install the certificate with cmd %s!" % certutil_cmd
+        print "Details refers: %s" % err
+        exit_status = -1
+
+
+# Install the specific driver
+def install_driver(driver_path, device_name, vol_utils):
+    install_driver_cmd = "C:\dpinst.exe /A /PATH %s /C /LM /Q /F" % driver_path
+    certutil(vol_utils)
+    try:
+        os.system(install_driver_cmd)
+        verify_driver_ver(driver_path, device_name)
+    except Exception, err:
+        print "Driver installation failed with cmd %s!" % install_driver_cmd
+        print "Details refers: %s and dpinst.log" % err
+        exit_status = -1
+
+
+# Uninstall the specific driver
+def uninstall_driver(driver_name):
+    driver_store = "C:\Windows\System32\DriverStore\FileRepository"
+    uninstall_driver_cmd = "C:\dpinst.exe /U %s /C /LM /Q /F"
+    for folder in os.listdir(driver_store):
+        if driver_name in folder:
+            driver_path = os.path.join(driver_store, folder)
+            for file in os.listdir(driver_path):
+                if ".inf" in file:
+                    inf_file = os.path.join(driver_path, file)
+                    uninstall_driver_cmd = uninstall_driver_cmd % inf_file
+                    try:
+                        os.system(uninstall_driver_cmd)
+                    except Exception, err:
+                        print "Driver uninstallat failed with cmd %s!" % uninstall_driver_cmd
+                        print "Details refers: %s and dpinst.log" % err
+                        exit_status = -1
+
+
+# Get the specific current driver version
+def get_current_driver_ver(device_name):
+    get_driver_ver_cmd = "wmic path win32_pnpsigneddriver where Devicename='%s' get driverversion" % device_name
+    key = "\d*\.\d*\.\d*\.\d*"
+    driver_version = os.popen(get_driver_ver_cmd).read()
+    return re.findall(key, driver_version, re.M)[-1]
+
+
+# Get expected driver version
+def get_expected_driver_ver(driver_path):
+    key = "\d*\.\d*\.\d*\.\d*"
+    for file in os.listdir(driver_path):
+        if ".inf" in file:
+            file_path = os.path.join(driver_path, file)
+            inf_file = open(file_path)
+            driver_info = inf_file.readlines()
+            inf_file.close()
+            for line in driver_info:
+                if "DriverVer" in line:
+                    return re.findall(key, line, re.M)[-1]

```

Comment:
it's better to conver to int, there maybe blank after the string e.g "1000" != "1000    "